static/js/subs.js

```diff
@@ -727,6 +727,41 @@ $(function () {
         update_announce_stream_state();
     });
 
+    // Search People
+    $(document).on('input', '.add-user-list-filter', function (e) {
+        var users = people.get_rest_of_realm();
+
+        var user_list = $(".add-user-list-filter");
+        if (user_list === 0) {
+            return;
+        }
+        var search_term = user_list.expectOne().val().trim();
+        var search_terms = search_term.toLowerCase().split(",");
+        search_terms = _.map(search_terms, function (s) {
+            return s.trim();
+        });
+
+        _.each(users, function (user) {
+            var names = user.full_name.toLowerCase().split(/\s+/);
+            names = _.map(names, function (s) {
+                return s.trim();
+            });
+
+            _.any(search_terms, function (search_term) {
```

Comment:
I'd rewrite this whole block in a more functional style. That means return a true/false value depending on if there's a search match or not (with no side effects). You can then use that true/false value to update the CSS